x/profile/client/cli/tx.go

```diff
@@ -0,0 +1,158 @@
+
+			picture := viper.GetString(flagProfilePic)
+			cover := viper.GetString(flagProfileCover)
+			pictures := types.NewPictures(picture, cover)
+
+			name := viper.GetString(flagName)
+			surname := viper.GetString(flagSurname)
+			bio := viper.GetString(flagBio)
+
+			msg := types.NewMsgCreateProfile(name, surname, args[0], bio, &pictures, cliCtx.FromAddress)
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+
+	cmd.Flags().String(flagName, "", "Name of the profile")
+	cmd.Flags().String(flagSurname, "", "Surname of the profile")
+	cmd.Flags().String(flagBio, "", "Biography of the profile")
+	cmd.Flags().String(flagProfilePic, "", "Profile related profile picture")
+	cmd.Flags().String(flagProfileCover, "", "Profile related profile cover")
+
+	return cmd
+}
+
+// GetCmdEditProfile is the CLI command for editing an profile
+func GetCmdEditProfile(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit [previous_moniker] [[new_moniker]]",
+		Short: "Edit an existent profile",
+		Long: fmt.Sprintf(`
+Edit an existing profile specifying the previous moniker, new moniker, name, surname, bio, a profile picture and cover.
+Every data except previous_moniker are optional.
+
+E.g (with all the other optional fields)
+%s tx profile edit leoDiCap DiCapLeo \
+	--name "Leo" \
+	--surname "Di Cap" \
+	--bio "Hollywood actor. Proud environmentalist" \
+	--picture "https://profilePic.jpg"
+	--cover "https://profileCover.jpg"
+`, version.ClientName),
+		Args: cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
+
+			picture := viper.GetString(flagProfilePic)
+			cover := viper.GetString(flagProfileCover)
+			pictures := types.NewPictures(picture, cover)
+
+			name := viper.GetString(flagName)
+			surname := viper.GetString(flagSurname)
+			bio := viper.GetString(flagBio)
+
+			prevMoniker := args[0]
+			newMoniker := args[0]
+			if len(args) > 1 {
+				newMoniker = args[1]
+			}
+			msg := types.NewMsgEditProfile(prevMoniker, newMoniker, name, surname, bio, &pictures, cliCtx.FromAddress)
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+
+	cmd.Flags().String(flagName, "default", "Name of the profile")
+	cmd.Flags().String(flagSurname, "default", "Surname of the profile")
+	cmd.Flags().String(flagBio, "default", "Biography of the profile")
+	cmd.Flags().String(flagProfilePic, "default", "Profile related profile picture")
+	cmd.Flags().String(flagProfileCover, "default", "Profile related profile cover")
+
+	return cmd
+}
+
+// GetCmdDeleteProfile is the CLI command for deleting an profile
+func GetCmdDeleteProfile(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete [moniker]",
```

Comment:
I'm not sure why we need the user moniker. Couldn't we simply find his profile based on his address?

Reply:
yup